presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/AnalyzerClient.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.spi.PrestoException;
+
+import javax.inject.Inject;
+
+import static com.facebook.presto.spi.StandardErrorCode.UNSUPPORTED_ANALYZER_TYPE;
+import static com.facebook.presto.sql.analyzer.AnalyzerType.BUILTIN;
+import static com.facebook.presto.sql.analyzer.AnalyzerType.NATIVE;
+import static java.util.Objects.requireNonNull;
+
+//TODO: Better name?
+public class AnalyzerClient
+{
+    private QueryPreparer builtInQueryPreparer;
```

Comment:
the naming here is funny, it sounds like they are actually both built in, but one is java and one is cpp.  But also, if we want to make it really extensible, I wouldn't define either, and instead would add the interface to the spi and have it be something totally pluggable, but default to builtin.